src/experimental/HomeChainCoordinator.sol

```diff
@@ -0,0 +1,363 @@
+     */
+    function submitBlockAndSendMessage(
+        bytes calldata rawHeader,
+        bytes[] calldata intermediateHeaders,
+        bytes32 _btcTxnHash,
+        bytes32[] calldata _proof,
+        uint256 _index,
+        bytes calldata _psbtData,
+        bytes calldata _options,
+        address _refundAddress
+    ) external payable whenNotPaused nonReentrant onlyOwner {
+        // 0. Submit block header along with intermediate headers to light client
+        bytes32 blockHash = lightClient.submitRawBlockHeader(rawHeader, intermediateHeaders);
+        // 1. Get merkle root to validate txn
+        bytes32 merkleRoot = lightClient.getMerkleRootForBlock(blockHash);
+        // 2. Send message
+        _sendMessage(merkleRoot, _btcTxnHash, _proof, _index, _psbtData, _options, _refundAddress);
+    }
+
+    /**
+     * @dev Sends a cross-chain message with PSBT data.
+     * @param _blockHash The block hash to validate the transaction.
+     * @param _btcTxnHash The BTC transaction hash.
+     * @param _proof The proof for the transaction.
+     * @param _index The index of the transaction in the block.
+     * @param _psbtData The PSBT data to be processed.
+     * @param _options LayerZero message options.
+     * @param _refundAddress The address to refund in case of failure.
+     */
+    function sendMessage(
+        bytes32 _blockHash,
+        bytes32 _btcTxnHash,
+        bytes32[] calldata _proof,
+        uint256 _index,
+        bytes calldata _psbtData,
+        bytes calldata _options,
+        address _refundAddress
+    ) external payable whenNotPaused nonReentrant onlyOwner {
+        bytes32 merkleRoot = lightClient.getMerkleRootForBlock(_blockHash);
+        _sendMessage(merkleRoot, _btcTxnHash, _proof, _index, _psbtData, _options, _refundAddress);
+    }
+
+    /**
+     * @dev Internal function to send a cross-chain message with PSBT data.
+     * @param _merkleRoot The merkle root for the block.
+     * @param _btcTxnHash The BTC transaction hash.
+     * @param _proof The proof for the transaction.
+     * @param _index The index of the transaction in the block.
+     * @param _psbtData The PSBT data to be processed.
+     * @param _options LayerZero message options.
+     * @param _refundAddress The address to refund in case of failure.
+     */
+    function _sendMessage(
+        bytes32 _merkleRoot,
+        bytes32 _btcTxnHash,
+        bytes32[] calldata _proof,
+        uint256 _index,
+        bytes calldata _psbtData,
+        bytes calldata _options,
+        address _refundAddress
+    ) internal {
+        // 0. btcTxnHash generated from the psbt data being shared should be the same as the one passed
+        bytes32 txid = TxidCalculator.calculateTxid(_psbtData);
+        if (txid != _btcTxnHash) {
+            revert InvalidPSBTData();
+        }
+
+        // 1. Parse and validate PSBT data
+        BitcoinTxnParser.TransactionMetadata memory metadata = _validatePSBTData(_psbtData);
```

Comment:
not verifying the fee and p2tr existence. Again put this in the TODO if not been take care now